src/core/server/execution_context/execution_context_service.ts

```diff
@@ -154,7 +154,7 @@ export class ExecutionContextService
 
   private getAsHeader(): string | undefined {
     if (!this.enabled) return;
-    // requestId may not be present in the case of FakeRequest
+    // requestId may not be present if unspecified by the client or in the case of FakeRequest
     const requestId = this.requestIdStore.getStore()?.requestId ?? 'unknownId';
     const executionContext = this.contextStore.getStore()?.toString();
     const executionContextStr = executionContext ? `;kibana:${executionContext}` : '';
```

Comment:
(This is correlated with the change in `src/core/server/elasticsearch/client/cluster_client.ts`)

ATM we kinda have an inconsistency between when EC is enabled or not, as when EC is enabled, the `x-opaque-id` value will be coming from `EC.getAsHeader`, and when it's not, the value is generated within `ClusterClient.getScopedHeaders`

ATM (with the PR's current change), the value matrix looks like:

| EC enabled | request.opaqueId present | EC present |`x-opaque-id` sent to ES |
|-|-|-|-|
| true | true | true | `'${opaqueId};kibana:${executionContext}'` | 
| true | true | false | `'${opaqueId}'` | 
| true | false | true | `'unknownId;kibana:${executionContext}'` | 
| true | false | false | `'unknownId'` | 
| false | true | true | N/A | 
| false | true | false | `'${opaqueId}'` | 
| false | false | true | N/A | 
| false | false | false | `undefined` | 

I got a few questions here:

- is this `unknownId` default value for `opaqueId` acceptable?
  -  https://github.com/elastic/kibana/issues/120124#issuecomment-984511107 suggest potentially using `kibana` as the default value, but then, in case of populated execution context, the value would look like `kibana;kibana:${executionContext}`. Are we fine with this format? Do we just want to use `kibana:${executionContext}` in that case instead?
- When EC is disabled, the header is populated by `ClusterClient.getScopedHeaders`. ATM when `request.opaqueId` is not present, we're not sending any `x-opaque-id` header to ES. Do we want to change that to send `kibana` (or any constant value we agreed on from the previous point) instead?
- (follow-up of next point) Should we decide to have the EC service the single source of truth for the `x-opaque-id` header and have `getAsHeader` always return a value even when disabled
  - this would require the `requestIdStore` to always be populated even when EC is disabled, which could be a performance issue as the whole purpose of allowing to disable the EC was for performances reasons  

@mshustov @lukeelmers what do you think about this?

Reply:
>> ATM when request.opaqueId is not present, we're not sending any x-opaque-id header to ES. Do we want to change that to send kibana (or any constant value we agreed on from the previous point) instead?

>@pgomulka what is better for the depreciation service of Elasticsearch?

either way is good. As long as it is constant. If Kibana wants to send a constant `x-opaque-id-from-kibana` it would help identifying its requests

Reply:
> I used kibana static value to simplify parsing when an incoming request contains x-opaque-id. If you find it easier to use kibana;${executionContext}, we can change the format.

Let's not overcomplicate the PR for no good reasons. The current format is good enough to address https://github.com/elastic/kibana/issues/120124, let's not change it for now. We'll reevaluate when we'll use the `baggage` header anyway.